src/components/InventoryGroupDetail/InventoryGroupDetail.js

```diff
@@ -64,48 +60,52 @@ const InventoryGroupDetail = ({ groupId }) => {
     return () => {
       dispatch({ type: 'GROUP_DETAIL_RESET' });
     };
-  }, [canViewGroup]);
+  }, [canViewGroup, groupId, dispatch]);
 
   useEffect(() => {
     // if available, change ID to the group's name in the window title
     chrome?.updateDocumentTitle?.(`${groupName || groupId} - Workspaces`);
-  }, [data]);
+  }, [groupName, groupId, chrome]);
 
   // TODO: append search parameter to identify the active tab
 
   const [hasEdgeImages, setHasEdgeImages] = useState(false);
-  const EdgeParityEnabled = useFeatureFlag('edgeParity.inventory-list');
+  const edgeParityEnabled = useFeatureFlag('edgeParity.inventory-list');
 
   useEffect(() => {
-    if (EdgeParityEnabled) {
+    // determines the active tab opened by default
+    const lookUpAvailableSystems = async () => {
```

Comment:
It would make sense to extract these API calls out of the effect and be a discrete separate function. It would further make sense to have two functions like `getConventionalHostList` and `getImmutableHostList` that perform Api calls with the desired filters set.